agent/agentproc/syscaller_other.go

```diff
@@ -0,0 +1,24 @@
+//go:build !linux
+// +build !linux
+
+package agentproc
+
+import "syscall"
+
+func NewSyscaller() Syscaller {
+	return nopSyscaller{}
+}
+
+type nopSyscaller struct{}
+
+func (nopSyscaller) SetPriority(pid int32, priority int) error {
+	return nil
+}
+
+func (nopSyscaller) GetPriority(pid int32) (int, error) {
+	return 0, nil
+}
+
+func (nopSyscaller) Kill(pid int32, sig syscall.Signal) error {
+	return nil
```

Comment:
I think the code would be more robust if these functions returned an unimplemented error that was then checked up the stack.

For example, some future caller would expect that they could obvserve the effect of `SetPriority` in `GetPriority` and that `Kill` would have a whole host of side effects.